mirrord/agent/src/steal/http.rs

```diff
@@ -121,6 +134,122 @@ impl HttpVersion {
             Self::NotHttp
         }
     }
+
+    /// Handles the HTTP connection accordingly to its version.
+    ///
+    /// See [`filter_task`] for more details.
+    #[tracing::instrument(level = "trace")]
+    async fn serve_connection(
+        self,
+        mut stream: DefaultReversibleStream,
+        original_destination: SocketAddr,
+        connection_id: ConnectionId,
+        filters: Arc<DashMap<ClientId, Regex>>,
+        matched_tx: Sender<HandlerHttpRequest>,
+        connection_close_sender: Sender<ConnectionId>,
+    ) -> Result<(), HttpTrafficError> {
+        match self {
```

Comment:
Continuing previous conversation. So what I mean is, instead of having HttpVersion implementing serve_connection, can't you have each of the variants implementing serve_connection? Then you'd only have v1 code within the scope of the v1 enum variant, and same for v2.
I guess you'd need to move serve_connection into a trait and then use enum_dispatch to do it, like what Aviram did with ContainerRuntime?

Reply:
Oh I get it.

Yeah Rust enum types are not proper types (yet), so to do this we would need an inner type (at least):

```rust
enum HttpVersion {
    V1(V1),
    V2(V2),
}

impl ServeConnection for V1 { ... }
```

Something like this should be enough to make it work.

Reply:
Do you think it's the way to go though?

Reply:
Yup, the code looks more local now.

Reply:
Did not go as far as putting the struct inside the enum variants, felt a bit too much. What do you think?

Reply:
Yup looks good!